runtime/command-start/src/main/java/io/aklivity/zilla/runtime/command/start/internal/airline/ZillaStartCommand.java

```diff
@@ -109,18 +111,18 @@ public void run()
         }
 
         EngineConfiguration config = new EngineConfiguration(props);
-        if ("file".equals(config.configURL().getProtocol()))
+        URL configURL = config.configURL();
+        if ("file".equals(configURL.getProtocol()))
         {
-            if (config.configURL().toString().endsWith("zilla.yaml") &&
-                    Files.notExists(Paths.get(config.configURL().getPath())))
+            Path configPath = Paths.get(configURL.getPath());
+            if (configPath.endsWith("zilla.yaml") && Files.notExists(configPath))
             {
-                String jsonConfig = Paths.get(config.configURL().getPath()).getParent() != null ?
-                        Paths.get(config.configURL().getPath()).getParent() + "/zilla.json" : "zilla.json";
-                props.setProperty(ENGINE_CONFIG_URL.name(), "file:" + jsonConfig);
-                config = new EngineConfiguration(props);
+                String configJSON = String.format("file:%s", configPath.resolveSibling("zilla.json"));
+                props.setProperty(ENGINE_CONFIG_URL.name(), configJSON);
+                configPath = Paths.get(config.configURL().getPath());
                 System.out.println("zilla.yaml file not found, loading zilla.json instead");
             }
-            if (config.configURL().toString().endsWith(".json"))
+            if (configPath.getFileName().endsWith(".json"))
```

Comment:
Does this code actually cause the deprecation warning to be printed?
Javadoc for [java.nio.file.Path](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#endsWith-java.lang.String-) says that `endsWith(String)` is checking the path segment as a whole.

Hence we need the path segment name, to test with `String.endsWith(String)` instead.
This was in the previous snippet as `configPath.getFileName().getName(0).endsWith(".json")` but didn't make it into the PR.

Could you please double check?